src/Route.php

```diff
@@ -2,56 +2,36 @@
 
 namespace Yiisoft\Router;
 
+use InvalidArgumentException;
+use LogicException;
 use Yiisoft\Router\Middleware\Callback;
 use Psr\Http\Message\ResponseInterface;
 use Psr\Http\Message\ServerRequestInterface;
 use Psr\Http\Server\MiddlewareInterface;
 use Psr\Http\Server\RequestHandlerInterface;
 
+use function current;
+use function is_callable;
+use function next;
+use function reset;
+
 /**
  * Route defines a mapping from URL to callback / name and vice versa
  */
 final class Route implements MiddlewareInterface, RequestHandlerInterface
 {
+    private string $name;
+    /** @var string[] */
+    private array $methods;
+    private string $pattern;
+    private string $host;
```

Comment:
Host is nullable :) [See](https://github.com/yiisoft/router/blob/a71a39c7faf22cf2db8dfba01e8bfa23198b9ec7/src/Route.php#L262)

Reply:
[Can't agree with you](https://github.com/yiisoft/router/blob/a71a39c7faf22cf2db8dfba01e8bfa23198b9ec7/src/Route.php#L126). It can be null only while it is not set at all.

Reply:
@viktorprogger Which makes it nullable. You know that accessing  initialized property will produce an error? If there is a place where `null` is acceptable value of property, then this property should be nullable and initialized as `null` by default.

Reply:
@viktorprogger Do you see call method `host` [there](https://github.com/yiisoft/yii-demo/blob/master/src/Factory/AppRouterFactory.php)?

Reply:
Thnx, @rob006, didn't know that. Will be more carefull.